### PR TITLE
[DOCS] update changelog for 0.24.0

### PR DESCRIPTION
**Pull request by merobi-hub**
Signed-off-by: Michael Robinson <merobi@gmail.com>

### Problem

Changelog needs to be updated for 0.24.0.

### Solution

Updates changelog for release.

> **Note:** All database schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [ ] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,26 @@
 # Changelog
 
-## [Unreleased](https://github.com/MarquezProject/marquez/compare/0.23.0...HEAD)
+## [Unreleased](https://github.com/MarquezProject/marquez/compare/0.24.0...HEAD)
+
+## [0.24.0](https://github.com/MarquezProject/marquez/compare/0.23.0...0.24.0) - 2022-07-28
```
Comment (wslulciuc) [inline CHANGELOG.md:5]: @merobi-hub: We would only update the changelog for `0.24.0` right before the release. The `Unreleased` section can still contain a list of changes (that you've added in this PR) that have yet to be released without an official release.

Comment (merobi-hub) [reply CHANGELOG.md:5]: @wslulciuc Thanks. My thinking was that if I opened a PR ahead of time the changelog could be reviewed and revised ahead of the release if needed without losing time

Comment (wslulciuc) [reply CHANGELOG.md:5]: Ahh makes sense! I didn't want to merge the PR until the tag / release was out, but looks like we're scheduled for the release today, so let's get this merged!

Comment (wslulciuc) [reply CHANGELOG.md:5]: @merobi-hub We'll also want to add a recent fix by @collado-mike (see [`2053`](https://github.com/MarquezProject/marquez/pull/2053))

**Review by wslulciuc — APPROVED** [review]
LGTM, but you'll also want to add a recent fix by @collado-mike (see [`2053`](https://github.com/MarquezProject/marquez/pull/2053))

File: CHANGELOG.md (hunk 1 above)
Comment (merobi-hub) [reply CHANGELOG.md:5]: @wslulciuc PR 2053 has been added. Good to go?

**Comment by merobi-hub** [comment]
@wslulciuc 2053 has been added. Are we good to go?

**Comment by merobi-hub** [comment]
@wslulciuc wait, I should update the date

**Review by wslulciuc — APPROVED** [review]

**Pull request merged**